code/modules/projectiles/guns/energy/special.dm

```diff
@@ -152,15 +156,29 @@
 		..()
 
 // Tool procs, in case plasma cutter is used as welder
+// Plasma cutters do not toggle so they need to explicitly use charge
+/obj/item/gun/energy/plasmacutter/tool_start_check(mob/living/user, amount=0)
+	if(!use(amount)) //nearly everything calls with amount=0 explicitly, handed in use()
+		to_chat(user, "<span class='warning'>You need more charge to complete this task!</span>")
+		return FALSE
+	user.flash_act(light_intensity)

```

Comment:
add this to tool_check_callback as well so the user is flashed after the welding is finished. see above.

Reply:
tool_check_callback is called every delay tick (and calls tool_use_check) and I currently can't figure a way to only do a flash when delay runs out.